hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/DataNode.java

```diff
@@ -1079,6 +1084,22 @@ private String reconfSlowIoWarningThresholdParameters(String property, String ne
     }
   }
 
+  private String reconfStripedReconstructionParameters(String property, String newVal)
+      throws ReconfigurationException {
+    String result = null;
+    try {
+      if (property.equals(DFS_DN_EC_RECONSTRUCTION_THREADS_KEY)) {
+        int size = (newVal == null ? DFS_DN_EC_RECONSTRUCTION_THREADS_DEFAULT :
+            Integer.parseInt(newVal));
+        result = Long.toString(size);
```

Comment:
what are we doing here? converting newVal to Integer & then again converting into string? shouldn't we just check null & directly pass the string, rather than this conversion

Reply:
There is special case, if the string passed is null, the default value will be used. So there's an extra conversion from int to string.